include/api.php

```diff
@@ -3386,6 +3419,64 @@ function api_friendica_activity(&$a, $type) {
 	api_register_func('api/friendica/activity/unattendno', 'api_friendica_activity', true, API_METHOD_POST);
 	api_register_func('api/friendica/activity/unattendmaybe', 'api_friendica_activity', true, API_METHOD_POST);
 
+	/**
+	 * @brief Returns notifications
+	 *
+	 * @param App $a
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	*/
+	function api_friendica_notification(&$a, $type) {
+		if (api_user()===false) throw new ForbiddenException();
+		if ($a->argc!==3) throw new BadRequestException("Invalid argument count");
+		$nm = new NotificationsManager();
+		
+		$notes = $nm->getAll(array(), "+seen -date", 50);
+		return api_apply_template("<auto>", $type, array('$notes' => $notes));
+	}
+	
+	/**
+	 * @brief Set notification as seen and returns associated item (if possible)
+	 *
+	 * POST request with 'id' param as notification id
+	 * 
+	 * @param App $a
+	 * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+	 * @return string
+	 */
+	function api_friendica_notification_seen(&$a, $type){
+		if (api_user()===false) throw new ForbiddenException();
+		if ($a->argc!==4) throw new BadRequestException("Invalid argument count");
+		
+		$id = (x($_REQUEST, 'id') ? intval($_REQUEST['id']) : 0);
+		
+		$nm = new NotificationsManager();		
+		$note = $nm->getByID($id);
+		if (is_null($note)) throw new BadRequestException("Invalid argument");
+		
+		$nm->setSeen($note);
+		if ($note['otype']=='item') {
+			// would be really better with an ItemsManager and $im->getByID() :-P
+			$r = q("SELECT * FROM item WHERE id=%d AND uid=%d",
```

Comment:
I, personally, would prefer selecting only the needed fields. This makes queries faster and wastes fewer memory.